### PR TITLE
ESQL: Add version to docs

### PR DESCRIPTION
**Pull request by alex-spies**
Docs for https://github.com/elastic/elasticsearch/pull/106824.

Does not cover the [REST API specs](https://github.com/elastic/elasticsearch/blob/main/rest-api-spec/src/main/resources/rest-api-spec/api/esql.query.json) as these don't cover the request body.

### CODE REVIEW

File: docs/reference/esql/esql-version.asciidoc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+++++
+<titleabbrev>Language versions</titleabbrev>
+++++
+
+[discrete]
+[[esql-versions-released]]
+==== Released versions
+
+{esql} is released at version `2024.04.01.🚀`.
```
Comment (leemthompo) [inline docs/reference/esql/esql-version.asciidoc:12]:
```suggestion
* Version `2024.04.01.🚀`.
```

Sets up the bulleted list for additions in future

File: docs/reference/esql/esql-version.asciidoc [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+This is the only available version for now.
```
Comment (leemthompo) [inline docs/reference/esql/esql-version.asciidoc:13]:
```suggestion
```

nit: might be self-evident

File: docs/reference/esql/esql-version.asciidoc [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+[discrete]
+[[esql-versions-explanation]]
+=== How versions work
+
+{esql} versions are separate from Elasticsearch versions.
```
Comment (leemthompo) [inline docs/reference/esql/esql-version.asciidoc:19]:
```suggestion
{esql} language versions are independent of {es} versions.
```

File: docs/reference/esql/esql-version.asciidoc [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+Their goal is to make sure that queries that were once written
+remain valid even as Elasticsearch and {esql} evolve.
```
Comment (leemthompo) [inline docs/reference/esql/esql-version.asciidoc:21]:
```suggestion
Versioning the language ensures that your queries will always
remain valid for that version, independent of new {es} and {esql} releases.
```

nit: rewording

File: docs/reference/esql/esql-version.asciidoc [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+For instance, the following query
```
Comment (leemthompo) [inline docs/reference/esql/esql-version.asciidoc:23]: It doesn't read well to have a code sample mid-sentence, so I'd recommend moving lines 41 + 42 up here.

File: docs/reference/esql/esql-version.asciidoc [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+// TEST[setup:library]
+
+will remain valid, even if a future version of {esql} introduces
+syntax changes or changes how the used commands or functions work.
+
+A released {esql} version will not receive breaking changes and
+will remain supported until it is deprecated.
+Already released {esql} versions will still profit from new features,
+bug fixes and performance improvements as long as those do not
+require breaking changes.
```
Comment (leemthompo) [inline docs/reference/esql/esql-version.asciidoc:48]:
```suggestion
We won't make breaking changes to released {esql} versions and
versions will remain supported until they are deprecated.
New features, bug fixes, and performance improvements 
will be continue to be added to released {esql} versions, 
provided they do not involve breaking changes.
```

Comment (leemthompo) [reply docs/reference/esql/esql-version.asciidoc:48]: rewording for clarity, apologies if suggestion messes up line lengths 🙈

**Review by leemthompo — COMMENTED** [review]
A little copyediting from me :)

**Comment by alex-spies** [comment]
Thanks for your help @nik9000 !

**Pull request merged**